### PR TITLE
[Refact] Factor 2 for Hamevo spectral gap in GPSR 

### PR DESCRIPTION
**Pull request by chMoussa**
Related to #397 . A factor 2 for the spectral gap is needed and enables increasing the test precision, so it should be correct now.

### CODE REVIEW

File: qadence/blocks/utils.py [hunk 1]
```diff
@@ -278,6 +278,11 @@ def uuid_to_eigen(block: AbstractBlock) -> dict[str, Tensor]:
             if b.eigenvalues_generator.numel() > 0:
                 result[uuid] = b.eigenvalues_generator
 
+                # GPSR asssumes a factor 0.5 for differientation
```
Comment (Roland-djee) [inline qadence/blocks/utils.py:281]:
```suggestion
                # GPSR assumes a factor 0.5 for differentiation
```

File: qadence/blocks/utils.py [hunk 2]
```diff
@@ -263,7 +263,7 @@ def expression_to_uuids(block: AbstractBlock) -> dict[Expr, list[str]]:
     return expr_to_uuid
 
 
-def uuid_to_eigen(block: AbstractBlock) -> dict[str, Tensor]:
+def uuid_to_eigen(block: AbstractBlock, scale_hamevo_eigen: bool = False) -> dict[str, Tensor]:
```
Comment (Roland-djee) [inline qadence/blocks/utils.py:266]:
```suggestion
def uuid_to_eigen(block: AbstractBlock, rescale_eigenvals: bool = False) -> dict[str, Tensor]:
```

File: qadence/blocks/utils.py [hunk 3]
```diff
@@ -263,7 +263,7 @@ def expression_to_uuids(block: AbstractBlock) -> dict[Expr, list[str]]:
...
     """Creates a mapping between a parametric block's param_id and its' eigenvalues.
```
Comment (Roland-djee) [inline qadence/blocks/utils.py:267]: Maybe add args details in docstring ?

**Review by Roland-djee — APPROVED** [review]
Thanks @chMoussa. Looking good.

**Review by jpmoutinho — APPROVED** [review]

**Pull request merged**